Reflection_Engine/Query/EnumValues.cs

```diff
@@ -0,0 +1,45 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2019, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Reflection;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.Engine.Reflection
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        public static IEnumerable<T> EnumValues<T>(this T enumeration)
+        {
+            return Enum.GetValues(enumeration.GetType()).Cast<T>();
```

Comment:
Some comments:

 - I personally would like to see some description/input/output attributes but won't hold up the PR over them.
 - I tried this with plugging in some enum components in GH and it worked as expected, however, when I plugged in the `type` of the enum (using the type component), it throws an error because `type` is not an `enum`. When I plug in the enum component I can still change the drop down of that enum, which doesn't make a lot of sense when getting just the full list of enum options. So I wonder if it's worth attempting to get all the values from the type rather than the enum itself? For a better UX? Thoughts @al-fisher @adecler ?

Reply:
Yeah, the motivation to create this method is purely practical: I was asked by the end users to investigate how to 'explode' the dropdown into a list. Happy to refactor, as long as the functionality is in place.

Reply:
@FraserGreenroyd I've added a method that takes the type - now we have two for both scenarios.